runtime/lua/vim/filetype.lua

```diff
@@ -2489,7 +2489,10 @@ function M.match(args)
 
   -- Finally, check file contents
   if contents or bufnr then
-    contents = contents or M.getlines(bufnr)
+    if contents == nil then
+      contents = M.getlines(bufnr, 1, 100)
+      contents[#contents + 1] = M.getlines(bufnr, -1)
```

Comment:
All we need is:

```suggestion
      if #contents == 100 then
        contents[#contents + 1] = M.getlines(bufnr, -1)
      end
```

but also maybe change the first `getlines` to `99`?

Reply:
No, because that would still duplicate if the length of `contents` is exactly 100.

I don't think catching all corner cases is worth it, which makes handling this at all not worth it.

(Adding a comment is fine, though.)

Reply:
What about the edit?

Reply:
eh, still a bit cumbersome, but that's one way to do it without extracting the full buffer contents (which is a deal-breaker).

I'll come up with a better conditional using that function.

Reply:
`if vim.api.nvim_buf_line_count(bufnr) > #contents` ?

Reply:
I'm rather thinking `if line_count(bufnr) > 100 then ... else contents = getlines(bufnr)`.